aws/resource_aws_sagemaker_model_test.go

```diff
@@ -0,0 +1,577 @@
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfig(rName),
+			},
+
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_basic(t *testing.T) {
+	var model sagemaker.DescribeModelOutput
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfig(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo", &model),
+
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "name", "terraform-testacc-sagemaker-model-foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "primary_container.#", "1"),
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "primary_container.0.image",
+						"174872318107.dkr.ecr.us-west-2.amazonaws.com/kmeans:1"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "execution_role_arn"),
+					resource.TestCheckResourceAttrSet("aws_sagemaker_model.foo", "arn"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_tags(t *testing.T) {
+	var model sagemaker.DescribeModelOutput
+	rName := acctest.RandString(10)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckSagemakerModelDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccSagemakerModelConfigTags(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo", &model),
+					testAccCheckSagemakerTags(&model, "foo", "bar"),
+
+					resource.TestCheckResourceAttr(
+						"aws_sagemaker_model.foo", "name", "terraform-testacc-sagemaker-model-foo"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.%", "1"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.foo", "bar"),
+				),
+			},
+			{
+				Config: testAccSagemakerModelConfigTagsUpdate(rName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckSagemakerModelExists("aws_sagemaker_model.foo", &model),
+					testAccCheckSagemakerTags(&model, "foo", ""),
+					testAccCheckSagemakerTags(&model, "bar", "baz"),
+
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.%", "1"),
+					resource.TestCheckResourceAttr("aws_sagemaker_model.foo", "tags.bar", "baz"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccAWSSagemakerModel_primaryContainerModelDataUrl(t *testing.T) {
```

Comment:
Should I combine TestAccAWSSagemakerModel_primaryContainerModelDataUrl, TestAccAWSSagemakerModel_primaryContainerHostname, TestAccAWSSagemakerModel_primaryContainerEnvironment into a single test or leave as is?

Reply:
Let's leave it as is just so we can pinpoint any issues if they arise with that specific attributes

Reply:
Ack.